Source/Data/DataConnection.cs

```
@@ -220,39 +220,51 @@ static void OnTraceInternal(TraceInfo info)
 					break;
 
 				case TraceInfoStep.Error:
-				{
-					var sb = new StringBuilder();
+			{
+				var sb = new StringBuilder();
 
-					for (var ex = info.Exception; ex != null; ex = ex.InnerException)
+				for (var ex = info.Exception; ex != null; ex = ex.InnerException)
+				{
+					try
 					{
-						sb
-							.AppendLine()
+					sb
+						.AppendLine()
 							.AppendLine("Exception: {0}".Args(ex.GetType()))
 							.AppendLine("Message  : {0}".Args(ex.Message))
-							.AppendLine(ex.StackTrace)
+						.AppendLine(ex.StackTrace)
+						;
+				}
+					catch
+					{

```

fixed formatting and added better comment to describe issue.
In other words - AseException from sybase provider contains bug, where it throws exception, when you try to read Message property. Without handling it, it will mask original exception